tests/stdlib/thttpclient_standalone.nim

```diff
@@ -5,25 +5,42 @@ discard """
 import asynchttpserver, httpclient, asyncdispatch, strutils
 
 block: # bug #16436
-  proc startServer() {.async.} =
+  proc startServer(): AsyncHttpServer =
+    result = newAsyncHttpServer()
+    result.listen(Port(0))
+
+  proc runServer(server: AsyncHttpServer) {.async.} =
+    var killServer = false
+
     proc cb(req: Request) {.async.} =
       let headers = { "Content-length": "15"} # Provide invalid content-length
+      killServer = true
       await req.respond(Http200, "Hello World", headers.newHttpHeaders())
 
-    var server = newAsyncHttpServer()
-    await server.serve(Port(5555), cb)
+    while not killServer:
+      if server.shouldAcceptRequest():
+        await server.acceptRequest(cb)
+      else:
+        poll()
```

Comment:
Please rewrite this so it doesn't use `poll` inside an async proc. There should just be either a `waitFor`, `runForever` or `poll` at the top-level of this test. Anything else just creates a mess (for example in this case the async proc is paused on the `await server.acceptRequest` and so will never get polled again causing the program to quit early.

Reply:
also, why does the diff introduce `killServer`, poll etc? 
note that the version before this PR had this but only in another place; this PR could at least handle the hard-coded ports issue (without introducing a new poll), and in a 2nd PR (or same PR works too) address the removal of the pre-existing `poll`

Reply:
The poll comes directly from the example in the documentation.  
[https://nim-lang.org/docs/asynchttpserver.html#example](https://nim-lang.org/docs/asynchttpserver.html#example)
If it is bad practice, why is it reccomended?

Reply:
> The poll comes directly from the example in the documentation.

use devel docs, which don't use poll, nor hard-coded ports: https://nim-lang.github.io/Nim/asynchttpserver.html#acceptRequest%2CAsyncHttpServer%2Cproc%28Request%29

Reply:
> can't something like this work?
Yep, looks like it works fine (and is much simpler).  I've updated the pull request in accord with your suggestion.